### PR TITLE
Add configurations to ignore node modules

### PR DESCRIPTION
**Pull request by SAkshat**
For some reason, the exclude pattern in the mx-app tsconfig files is not being read. Rather than going through the code and figuring out the inner workings, I've added this quickfix to get our work on the go. This can be revisited in the future if we plan to keep using this functionality.

### CODE REVIEW

File: src/findCandidates.ts [hunk 1]
```diff
@@ -27,7 +27,11 @@ async function findCandidates() {
 
   const fileToImports = new Map<string, string[]>();
   for (const file of await forEachFileInSrc(srcRoot)) {
-    fileToImports.set(file, getImportsForFile(file, srcRoot))
+    if(!file.includes('node_modules')) {
+      fileToImports.set(file, getImportsForFile(file, srcRoot));
+    } else {
+      console.log(file)
```
Comment (adrienplg) [inline src/findCandidates.ts]: What's the point of console logging the file here?

File: src/tsHelper.ts [hunk 2]
```diff
@@ -10,13 +10,17 @@ export function getImportsForFile(file: string, srcRoot: string) {
...
+      if (fs.existsSync(index)) {
+        // https://basarat.gitbooks.io/typescript/docs/tips/barrel.html
+        console.warn(`Warning: Barrel import: ${path.relative(srcRoot, file)}`)
+        file = index
+      } else {
+        throw new Error(`Warning: Importing a directory without an index.ts file: ${path.relative(srcRoot, file)}`)
+      }
     } else {
-      throw new Error(`Warning: Importing a directory without an index.ts file: ${path.relative(srcRoot, file)}`)
+      return;
```
Comment (adrienplg) [inline src/tsHelper.ts]: This else condition seems unnecessary.

File: src/findCandidates.ts (hunk 1 above)
Comment (SAkshat) [reply src/findCandidates.ts]: This was mainly used for testing. I've removed logging statements added by me.

File: src/tsHelper.ts (hunk 2 above)
Comment (SAkshat) [reply src/tsHelper.ts]: Removed

File: src/tsHelper.ts [hunk 3]
```diff
@@ -10,13 +10,15 @@ export function getImportsForFile(file: string, srcRoot: string) {
...
-      // https://basarat.gitbooks.io/typescript/docs/tips/barrel.html
-      console.warn(`Warning: Barrel import: ${path.relative(srcRoot, file)}`)
-      file = index
-    } else {
-      throw new Error(`Warning: Importing a directory without an index.ts file: ${path.relative(srcRoot, file)}`)
+    if(!file.includes('node_modules')) {
+      const index = path.join(file, "index.ts")
+      if (fs.existsSync(index)) {
+        // https://basarat.gitbooks.io/typescript/docs/tips/barrel.html
+        console.warn(`Warning: Barrel import: ${path.relative(srcRoot, file)}`)
```
Comment (adam-zethraeus) [inline src/tsHelper.ts:17]: is this pattern of grouping bad?

**Review by adam-zethraeus — APPROVED** [review]

**Comment by adam-zethraeus** [comment]
@SAkshat could you make Adrien and myself admins on this repo?

**Comment by SAkshat** [comment]
> @SAkshat could you make Adrien and myself admins on this repo?

@adrienplg , @adam-zethraeus - Done.

**Comment by adam-zethraeus** [comment]
thanks!

also: it looks like you *can* do a whitelist with just tsc (and you did!) awesome. I thought that was part of what this script did. Nice that we don't have to pull it in as the submodule then!

**Review by adrienplg — APPROVED** [review]

**Pull request merged**